src/Symfony/Bundle/SecurityBundle/Security.php

```diff
@@ -101,13 +102,14 @@ public function getFirewallConfig(Request $request): ?FirewallConfig
     }
 
     /**
-     * @param UserInterface $user              The user to authenticate
-     * @param string|null   $authenticatorName The authenticator name (e.g. "form_login") or service id (e.g. SomeApiKeyAuthenticator::class) - required only if multiple authenticators are configured
-     * @param string|null   $firewallName      The firewall name - required only if multiple firewalls are configured
+     * @param UserInterface    $user              The user to authenticate
+     * @param string|null      $authenticatorName The authenticator name (e.g. "form_login") or service id (e.g. SomeApiKeyAuthenticator::class) - required only if multiple authenticators are configured
+     * @param string|null      $firewallName      The firewall name - required only if multiple firewalls are configured
+     * @param BadgeInterface[] $badges            Badges to add to the user's passport
      *
      * @return Response|null The authenticator success response if any
      */
-    public function login(UserInterface $user, string $authenticatorName = null, string $firewallName = null): ?Response
+    public function login(UserInterface $user, string $authenticatorName = null, string $firewallName = null /* array $badges = [] */): ?Response
```

Comment:
As this class is `@final`, we should be able to add the argument natively without such BC layer

Reply:
Oh okay